### PR TITLE
[RateLimiter][Security] (obsolete) Add an easy way to customize "interval" in the default security.login_throttling

### PR DESCRIPTION
**Pull request by damienfa**
| Q             | A
| ------------- | ---
| Branch       | 5.2
| Bug fix      |  no
| New feature  | yes
| Deprecations | no
| License       | MIT
| Doc PR        | ⚠️  no doc

The only way to customize the default rate-limiter's options of the login_throttling (means fixed_window / 1 minute / 5 tokens) are through a custom limiter, which implies to declare a rate-limiter factory in the "framework.rate_limiter", a service which use this factory etc. It's really heavy just for changing an interval (moreover, 1 minute can be discutable).
 
In this PullRequest, I just propose to allow an `interval` option. 

Example : 
```yaml
security:
  firewalls:
    main:
       login_throttling:
           max_attempts: 5
           interval: '15 minutes'
```

⚠️   At the time of writing, there is no Symfony documentation about the Login Throttling feature ([only a blog post](https://symfony.com/blog/new-in-symfony-5-2-login-throttling)). Is there a reason to that ? Or may I propose something ?

### CODE REVIEW

**Comment by wouterj** [comment]
Hi @damienfa! Thanks for your time creating this PR. I think such an option makes lots of sense!

Are you willing to add a test for this new feature? (I can help if you are unsure how to do this) Also note the failing CS check in "fabbot.io" is related to this PR (there seems to be a trailing whitespace). At last, can you please add a small note to `src/Symfony/SecurityBundle/CHANGELOG.md` to say you introduce this new option?

> At the time of writing, there is no Symfony documentation about the Login Throttling feature (only a blog post). Is there a reason to that ? Or may I propose something ?

Lacking time was the main factor (and we have to be careful to mention it only works when using experimental authenticators). I'm happy to review if you propose some docs for this feature.

**Comment by damienfa** [comment]
Ok, thanks for your answer.
I will take some time this week-end to fix the failing CS check (fabbot.io)
and try to create some unit-test. Is there a test concerning the
login_throttling feature or configuration ? I haven't find anything about
that in the existing tests (and it's why I didn't implement it).
Thanks,

**Comment by wouterj** [comment]
Hi! We have a functional test for login throttling here: https://github.com/symfony/symfony/blob/5.x/src/Symfony/Bundle/SecurityBundle/Tests/Functional/FormLoginTest.php#L115 (which uses this config: https://github.com/symfony/symfony/blob/5.x/src/Symfony/Bundle/SecurityBundle/Tests/Functional/app/StandardFormLogin/login_throttling.yml)

**Comment by damienfa** [comment]
Hello @wouterj , as discussed I've added some functional test (improving the existing one [here](https://github.com/damienfa/symfony/commit/9de6edf8d5bd40083f23cf4bfd36ac0637e65892) with an interval and two more login attempts). Also, [I've fixed](https://github.com/symfony/symfony/pull/39927/commits/752da9b70fa31c02379d3a914bf5fdc6c69c84d1) the fabbot.io check (it was due to a trailing space). 

However, the Integration test (7.4) is now failing, and I don't understand which relation it has with my commit 🤷🏻‍♂️   Any help ?

**Comment by wouterj** [comment]
No worries! The failures also occur in the `5.x` branch: https://github.com/symfony/symfony/runs/1840808458 So they are certainly not your fault :) 

I'll checkout this PR somewhere the coming days. Thanks for your update!

File: src/Symfony/Bundle/SecurityBundle/CHANGELOG.md [hunk 1]
```diff
@@ -1,6 +1,11 @@
 CHANGELOG
 =========
 
+dev
```
Comment (wouterj) [inline src/Symfony/Bundle/SecurityBundle/CHANGELOG.md]:
```suggestion
5.3
```

File: src/Symfony/Bundle/SecurityBundle/CHANGELOG.md [hunk 2]
```diff
@@ -1,6 +1,11 @@
...
+---
+
+ * Added a `login_throttling.interval` (in `security.firewalls`) to permits easily to change the throttling interval.
```
Comment (wouterj) [inline src/Symfony/Bundle/SecurityBundle/CHANGELOG.md]:
```suggestion
 * Add a `login_throttling.interval` (in `security.firewalls`) option to change the default throttling interval.
```

(not really important, but for any future contributions: we recently changed/formalized changelog conventions, see https://symfony.com/doc/current/contributing/code/conventions.html#writing-a-changelog-entry )

**Review by maxhelias — APPROVED** [review]

**Review by wouterj — APPROVED** [review]
2 minor comments, looks good otherwise!

File: src/Symfony/Bundle/SecurityBundle/Tests/Functional/FormLoginTest.php [hunk 3]
```diff
@@ -112,7 +112,7 @@ public function testFormLoginRedirectsToProtectedResourceAfterLogin(array $optio
      * @dataProvider provideInvalidCredentials
      * @group time-sensitive
      */
-    public function testLoginThrottling($username, $password)
+    public function testLoginThrottling(string $username, string $password, int $attempt_index)
```
Comment (Nyholm) [inline src/Symfony/Bundle/SecurityBundle/Tests/Functional/FormLoginTest.php]: Let's use `$attemptIndex` instead.

**Review by Nyholm — CHANGES_REQUESTED** [review]
Thank you for this PR. This is great!

Could you fix the minor things so this can be merged in 5.3?

**Comment by Nyholm** [comment]
Thank you. 

1. Fabbot needs to be happy. If you rebase your PR on branch 5.x, it should be green. 
2. This is a feature, we do need a changelog entry on the SecurityBundle.

**Comment by damienfa** [comment]
Sorry, I have to close this merge request. Following [this comment](https://github.com/symfony/symfony/pull/39927#issuecomment-787675178) I needed to rebase on branch 5.x. 
Unfortunately, the "rebase" goes a way I've never experienced (it runs in a loop of conflicts and useless merge). 
To do this, I simply created [a new PR (see here)](https://github.com/symfony/symfony/pull/40339) from this one. It's more readable, clearer.

**Pull request closed without merging**